thetis/momentum_eq.py

```diff
@@ -115,6 +214,26 @@ def residual(self, solution, solution_old, fields, fields_old, bnd_conditions=No
 
 
 class HorizontalAdvectionTerm(MomentumTerm):
+    r"""
+    Horizontal advection term, :math:`\nabla_h \cdot (\textbf{u} \textbf{u})`
+
+    The weak form reads
+
+    .. math::
+        \int_\Omega \nabla_h \cdot (\textbf{u} \textbf{u}) \cdot \boldsymbol{\psi} dx
+        = - \int_\Omega \nabla_h \boldsymbol{\psi} : (\textbf{u} \textbf{u}) dx
+        + \int_{\mathcal{I}_h \cup \mathcal{I}_v} \textbf{u}^{\text{up}} \cdot
+        \text{jump}(\boldsymbol{\psi} \textbf{n}_h) \cdot \text{avg}(\textbf{u}) dS
+
+    where the right hand side has been integrated by parts; :math:`:` stand for
+    the Frobenius inner product, :math:`\textbf{n}_h` is the horizontal
+    projection of the normal vector, :math:`\textbf{u}^{\text{up}}` is the
+    upwind value, and :math:`\text{jump}` and :math:`\text{avg}` denote the
+    jump and average operators across the interface.
+    If :math:`\bar{\textbf{u}}` belongs to a discontinuous function space, the
+    latter form is used.
```

Comment:
I'm not sure I understand this sentence.  Do you mean that if u_bar is discontinuous, it is upwinded?